cmake/jana_plugin.cmake

```diff
@@ -90,6 +100,8 @@ macro(plugin_add _name)
             EXPORT EICreconTargets
             DESTINATION ${PLUGIN_LIBRARY_OUTPUT_DIRECTORY}
         )
+        # Link library into interface library
+        target_link_libraries(${PROJECT_NAME} INTERFACE ${_name}_library)
```

Comment:
It would simpler to instead move
```
        target_include_directories(${_name}_plugin
          PUBLIC
            $<BUILD_INTERFACE:${EICRECON_SOURCE_DIR}/src>
            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>
        )
```
to apply to the `${PROJECT_NAME}`? The current way with indirection looks fancy, but none of the plugins/libraries present different interfaces, I reckon.

Reply:
While we can add this to the PROJECT_NAME target, we would still need to add an install interface link library to the granular targets that are exported or they would be invalid and downstream would complain if it includes a build interface location. We can move away from exporting the plugins and libraries as targets but that's more complicated with the way it's set up now.